upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -127,6 +127,31 @@ TOKEN=$(curl -s -S -d grant_type=client_credentials \
   https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
 export TOKEN
 
+# Takes as input the path to a CSM RPM in the expanded tarball. e.g.:
+# /etc/cray/upgrade/csm/csm-1.5.0-beta.64/tarball/csm-1.5.0-beta.64/rpm/cray/csm/noos/x86_64/hpe-csm-virtiofsd-1.7.0-hpe1.x86_64.rpm
+# /etc/cray/upgrade/csm/csm-1.5.0-beta.64/tarball/csm-1.5.0-beta.64/rpm/cray/csm/noos/noarch/csm-testing-1.16.59-1.noarch.rpm
+# /etc/cray/upgrade/csm/csm-1.5.0-beta.64/tarball/csm-1.5.0-beta.64/rpm/cray/csm/sle-15sp4/x86_64/loftsman-1.2.0-3.x86_64.rpm
+#
+# And outputs the corresponding Nexus URL of that RPM. e.g.:
+# https://packages.local/repository/csm-noos/x86_64/hpe-csm-virtiofsd-1.7.0-hpe1.x86_64.rpm
+# https://packages.local/repository/csm-noos/noarch/csm-testing-1.16.59-1.noarch.rpm
+# https://packages.local/repository/csm-sle-15sp4/x86_64/loftsman-1.2.0-3.x86_64.rpm
+#
+# Exits non-0 if the specified path does not start with / and end with /rpm/cray/csm/<os-string>/<arch>/<rpm-name>.rpm
+function csm_rpm_tarball_path_to_nexus_url {
+  if [[ $# -ne 1 ]]; then
+    echo "ERROR: $0: This function requires exactly 1 argument but received $#. Invalid arguments: $*" >&2
+    return 1
+  elif [[ -z $1 ]]; then
+    echo "ERROR: $0: Argument to this function may not be blank" >&2
+    return 1
+  elif [[ ! $1 =~ ^/.*/rpm/cray/csm/[^/[:space:]]+/[^/[:space:]]+/[^/[:space:]]+[.]rpm$ ]]; then
+    echo "ERROR: $0: RPM path not in expected format: '$1'" >&2
```

Comment:
My $0.02.. not important for holding the PR up, but if you're saying "ERROR:" I'd encourage routing these `echo`s to `stderr`.